src/calculate-deal.mo

```diff
@@ -0,0 +1,79 @@
+import Iter "mo:base/Iter";
+import Nat "mo:base/Nat";
+
+module {
+
+  public func calculateDeal<Order>(
+    asks : Iter.Iter<Order>,
+    bids : Iter.Iter<Order>,
+    getVolume : (Order) -> Nat,
+    getPrice : (Order) -> Float,
+  ) : (
+    asksAmount : Nat,
+    bidsAmount : Nat,
```

Comment:
Maybe call this `asksLen` for "length of lists of asks that are filled". Or `nAsks`.

"Amount" can be confused with volume, and is meant to mean "number" here.

Reply:
renamed